aten/src/ATen/native/native_functions.yaml

```diff
@@ -1473,6 +1473,24 @@
 
 - func: layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True) -> Tensor
 
+- func: nanprod(Tensor self, *, ScalarType? dtype=None) -> Tensor
+  variants: function, method
+  supports_named_tensor: True
+
+- func: nanprod.dim_int(Tensor self, int dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
+  variants: function, method
+  supports_named_tensor: True
+
+- func: nanprod.int_out(Tensor self, int dim, bool keepdim=False, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
```

Comment:
I was surprised to see that prod.int_out exists, since our documentation doesn't describe prod as having an out kwarg: https://pytorch.org/docs/master/generated/torch.prod.html?highlight=prod#torch.prod. Would you correct its documentation, too?

Reply:
Sure, Ruberry! I'll add this as my next commit once all tests are done running.